### PR TITLE
Update feature

### PR DESCRIPTION
**Pull request by Ali-0111**
- Before updating `species` and using `transaction`, the table initial state:

image

============================================================================
-Transaction started by : ( BEGIN TRANSACTION; )
image

- Inside Transaction update applied and verified usind SELECT * FROM:

image

- After using ROLLBACK, table changed to initial state:

image

- Adding `records` for `species` column:

image

- After COMMIT:

image
============================================================================
- [ ]  Delete all animals born after Jan 1st, 2022 and -1 multiplied to weight_kg field.
image

image

image

=================================================================
- [ ] Number of Available Animals:

image

- [ ] Number of Animals , which did not escape 
image

- [ ] AVG value of Weight_kg

image

- [ ] Animal most escaped neutered or not neutered:
image

- [ ] Max and Min value for weight in each TYPE:
image

- [ ] The average number of escape attempts per animal type of those born between 1990 and 2000
image

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -16,3 +16,34 @@ INSERT INTO animals
...
+(name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES
+('Boarmon', '2005-06-7', 20.4, true, 7 );
+
+INSERT INTO animals 
+(name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES
+('Blossom', '1998-10-13', 17, true, 3 );
+
+INSERT INTO animals 
+(name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES
+('Ditto', '2022-05-14', 22, true, 4 );
```
Comment (mckent05) [inline data.sql]: - Good job done so far, but you seem to have switched values for the weights and escape attempt attributes, please make sure you properly and carefully input data as given in the [project requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md) 👍 👍

File: data.sql [hunk 2]
```diff
@@ -16,3 +16,34 @@ INSERT INTO animals
...
+(name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES
+('Charmander', '2020-02-08', 0, false, 11);
+
+INSERT INTO animals 
+(name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES
+('Plantmon', '2021-11-15', 2, false, 5.7);
+
+INSERT INTO animals 
+(name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES
+('Squirtle', '1993-04-2', 3, false, 12.13);
```
Comment (mckent05) [inline data.sql:29]: - Good job done so far, but the weights for `Charmander`, `Plantmon`, `Squirtle`, and `Angemon`, please fix this so you could have the right query result 👍 👍

File: queries.sql [hunk 3]
```diff
@@ -8,4 +8,68 @@ select * from animals where name IN ('Agumon', 'Pikachu');
...
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT delete;
+
+UPDATE animals
+SET weight_kg = -1 * weight_kg;
+
+SELECT * FROM animals;
+ROLLBACK;
+COMMIT;
```
Comment (mckent05) [inline queries.sql]: - After correcting the data in your `data.sql` file, please re-run the transaction below again, as I noticed you skipped some in your file 🙏 
Inside a transaction:
- Delete all animals born after Jan 1st, 2022.
- Create a savepoint for the transaction.
- Update all animals' weight to be their weight multiplied by -1.
- Rollback to the savepoint
- Update all animals' weights that are negative to be their weight multiplied by -1.
- Commit transaction

File: queries.sql [hunk 4]
```diff
@@ -8,4 +8,68 @@ select * from animals where name IN ('Agumon', 'Pikachu');
...
+
+-- Requirement-4 
+SELECT count(name) as Number_of_Animals from animals;
+SELECT name, escape_attempts FROM animals WHERE escape_attempts = 0; 
+SELECT AVG(weight_kg) AS Weight_Average  FROM animals;
```
Comment (mckent05) [inline queries.sql:74]: - Please re-run this query and attach the right results in your PR after correcting the data in your `data.sql` file. Please fix this 👍 👍

File: queries.sql [hunk 5]
```diff
@@ -8,4 +8,68 @@ select * from animals where name IN ('Agumon', 'Pikachu');
...
+SELECT name, neutered, escape_attempts 
+FROM animals
+WHERE escape_attempts = (SELECT MAX(escape_attempts) FROM animals);
```
Comment (mckent05) [inline queries.sql]: - this is a wrong query, to get the animal that has escaped the most, neutered or non-neutered, please use the query below:
   `SELECT neutered, AVG(escape_attempts) FROM animals
   GROUP BY neutered`

File: queries.sql [hunk 6]
```diff
@@ -8,4 +8,68 @@ select * from animals where name IN ('Agumon', 'Pikachu');
...
+-- Requirement-4 
+SELECT count(name) as Number_of_Animals from animals;
+SELECT name, escape_attempts FROM animals WHERE escape_attempts = 0; 
+SELECT AVG(weight_kg) AS Weight_Average  FROM animals;
+SELECT name, neutered, escape_attempts 
+FROM animals
+WHERE escape_attempts = (SELECT MAX(escape_attempts) FROM animals); 
+
+SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;
+
```
Comment (mckent05) [inline queries.sql:82]: - Please re-run this query and attach the right results in your PR after correcting the data in your data.sql file. Please fix this 👍 👍
- Also, I cannot see the query for this query, `What is the average number of escape attempts per animal type of those born between 1990 and 2000?` in your query file, though I see you have included the results in your PR. Please fix this 🙏 
- Also, the attached result for the above-mentioned query is not correct, to get the `average number of escape attempts per animal type of those born between 1990 and 2000?`. Please use this query:
```sql
SELECT species, AVG(escape_attempts) FROM animals
   GROUP BY species
   WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/12/31'
```

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Ali-0111, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job updating your animals table 👏 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ali-0111** [comment]
@ CR, 
- I inserted the value again, and re ran the queries.

**Review by divinecharlotte — APPROVED** [review]
# Hello @Ali-0111      😃 🥳 🎉 🎈 

Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say  👍🏻 ✅ 
### To Highlight 💯


- Good implementation of the required changes👍 
- Your PR is professional 👍 👍
 




## Congratulations! 🎉 



Cheers, and Happy coding!👏👏👏


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ali-0111** [comment]
Thanks for the review.

**Pull request merged**